spec/jobs/risc_delivery_job_spec.rb

```diff
@@ -64,23 +64,44 @@
       )
     end
 
-    context 'SSL network errors' do
+    shared_examples 'retries and logs event when attempts exhausted' do |exception|
       before do
-        stub_request(:post, push_notification_url).to_raise(Faraday::SSLError)
+        stub_request(:post, push_notification_url).to_raise(exception)
+        allow_any_instance_of(described_class).to receive(:analytics).and_return(job_analytics)
       end
 
-      it 'raises and retries via ActiveJob' do
-        expect { perform }.to raise_error(Faraday::SSLError)
-      end
-
-      context 'it has already failed twice' do
-        before do
-          allow(job).to receive(:executions).and_return 2
+      context 'when the job fails for the 1st time' do
+        it 'raises and retries via ActiveJob' do
+          expect do
+            RiscDeliveryJob.new.perform(
+              push_notification_url: push_notification_url,
+              jwt: jwt,
+              event_type: event_type,
+              issuer: issuer,
+            )
+          end.to raise_error(exception)
+
+          expect(job_analytics).not_to have_logged_event(
+            :risc_security_event_pushed,
+            risc_event_payload.merge(
+              error: 'Exception from WebMock',
+            ),
+          )
         end
+      end
 
+      context 'when the job fails past the configured retry attempts' do
         it 'logs an event' do
-          expect { perform }.to_not raise_error
-
+          perform_enqueued_jobs do
```

Comment:
i'm not familiar with how ActiveJob::TestHelper works -- can you clarify how this is having the job fail twice? thanks!

Reply:
It is not documented well in https://api.rubyonrails.org/v7.1.3.4/classes/ActiveJob/TestHelper.html#method-i-perform_enqueued_jobs, but `perform_enqueued_jobs` does wait till all jobs that resulted in retries are also "performed".

Reply:
thanks!